godot-macros/src/docs.rs

```diff
@@ -0,0 +1,151 @@
+use crate::class::{ConstDefinition, FuncDefinition};
+use proc_macro2::TokenStream;
+use quote::ToTokens;
+use regex::Regex;
+use std::sync::OnceLock;
+use venial::*;
+
+/// Converts [markdown](https://en.wikipedia.org/wiki/Markdown) to [BBCode](https://en.wikipedia.org/wiki/BBCode).
+pub fn md2bb(md: &str) -> String {
+    macro_rules! r {
+        ($($expr:literal => $replace:literal)+) => {{
+            const REGEXES: OnceLock<[(Regex, &str); 6]> = OnceLock::new();
+            REGEXES.get_or_init(|| [
+                $((Regex::new($expr).unwrap(), $replace)),+
+            ])
+        }}
+    }
+    // this is not the most robust way
+    let mut out = md.to_string();
+    for (search, replace) in r![
+        "_([^_]+)_" => "[i]$1[/i]"
+        "\\*\\*([^\\*\\*]+)\\*\\*" => "[b]$1[/b]"
+        "\\*([^\\*]+)\\*" => "[i]$1[/i]"
+        "```([^`]+)```" => "[codeblock]$1[/codeblock]"
+        "`([^`]+)`" => "[code]$1[/code]"
+        "~~([^~]+)~~" => "[s]$1[/s]"
+    ] {
+        out = search.replace_all(&out, *replace).into_owned();
+    }
+    out
```

Comment:
i really dont think this is the approach we should go with, a while ago i think @Bromeon and i discussed using the [markdown](https://lib.rs/crates/markdown) crate to parse the markdown, which i think would be a better approach. though probably behind a feature flag so people who dont care about generating documentation can disable it